Add sentry

katemildner opened the pull request:
Added and configured Sentry, as well as manually tested if connection is working properly.

katemildner committed:
Add Sentry gem for error monitoring

katemildner committed:
Add and config Figaro gem for ENV managing

katemildner committed:
Setup Sentry

katemildner committed:
Fix 'sentry_key' variable

luktuc commented on config/initializers/sentry.rb:
```diff
@@ -0,0 +1,13 @@
+  config.traces_sampler = lambda do |context|
+    true
+  end
+end
```
Seems like rubocop missed this file :)

katemildner replied:
Yes, so far I added basic rubocop and I plan to add rubocop-rails, rubocop-rspec and rubocop-performance and check this code thoroughly on a separate branch before I start writing a lot of my own. Would that be ok? 🤔

luktuc replied:
Should be. As Github suggested it should be a new line and the end of the file and rubocop should warn about this. If you to let rubocop to lint your files you can use `rubocop -a` command 🤔

katemildner replied:
I think everything is in order now - can I merge this PR? :)

jandudulski commented on config/initializers/sentry.rb:
```diff
@@ -0,0 +1,13 @@
+Sentry.init do |config|
+  config.dsn = ENV['sentry_key']
```
Sentry by default takes dsn from `SENTRY_DSN` key and I would recommend to stay with the default.

katemildner replied:
You're right, thank you.

jandudulski commented on config/initializers/sentry.rb:
```diff
@@ -0,0 +1,13 @@
+  # of transactions for performance monitoring.
+  # We recommend adjusting this value in production
+  config.traces_sample_rate = 1.0
+  # or
```
So which version do you prefer? ;)

katemildner replied:
Why not bo-... Okay, missed this, thanks. ;)

jandudulski commented on Gemfile:
```diff
@@ -27,6 +27,13 @@ gem 'bootsnap', '>= 1.4.4', require: false
+gem 'sentry-rails', '~> 4.6.4'
+
+# Set Rails-only ENV with Figaro
+gem 'figaro', '~> 1.2'
```
Whoa. Haven't seen figaro for a long time :D While it's probably still fine, most people use [dotenv](https://github.com/bkeepers/dotenv) this days or just [Rails credentials](https://edgeguides.rubyonrails.org/security.html#environmental-security)

katemildner replied:
Noted! Figaro was the first thing I found, and last version came out _just over a year ago_, so I've decided it's fine. I'll leave it for now and if needed I'll upgrade to one of the above.

katemildner committed:
fix: Senrty config file according to suggestions in PR

Changed sentry_key to SENTRY_DNS and manually tested; chose one way
to trace samples.

katemildner committed:
add .rubocop.yml to .gitignore

katemildner committed:
fix: resolve last offenses found by rubocop

Resolved remaining offenses found by Rubocop and excluded the out-of-the-box files generating
offenses that won't be fixed due to their complexity: app/models/application_
record.rb; config/application.rb; bin/bundle; test/**/*.

katemildner committed:
add rubocop-rails

katemildner committed:
fix: resolve offenses found by rubocop-rails

wmaciejak commented on .gitignore:
```diff
@@ -32,3 +32,6 @@
 /config/application.yml
+
+# Ignore configuration of Rubocop(s)
+.rubocop.yml
```
Why do you want to ignore rubocop configuration?

Storing rubocop configuration for the whole project is important cause we want to have consistent behavior in all environments for all developers who work on the project. 
Consider that I've cloned your project and ran rubocop in this branch and I see that there we have:
1. 9 warnings
2. 1 error (it's a syntax error!)
3. 1 correctable warning (so it might be automaticall solved)
![image](https://user-images.githubusercontent.com/4128525/129086526-e17d2022-45e4-47e1-89c3-ae3997fd977d.png)

I see this status cause probably you did not upload your updated configuration of rubocop. It's a good idea to not add to git our local text editor settings, but it's an awesome idea to always add to git configuration of tools (or gems) that we using in the project.

Also, consider that you can ignore in rubocop configuration all files from directory `bin`, you can easily remove at all `test` directory and then you will have only 2 warnings, where both might be easily solved -> [article](https://thedevpost.com/blog/fix-rubocop-missing-comment/)

katemildner replied:
I have to admit - I didn't think this through - or rather I've had the wrong idea about rubocop, treating it as my local tool.

As you've already guessed - everything in here is handled by my local rubocop configuration file.
I also do not care right now about the `test` directory, because it is redundant and it will be replaced by rspec anyway, but I figured it's not the place to deal with it (since it's a branch for Sentry configuration - although I've been fixing unrelated rubocop offenses here as well...).

I'm also aware of one more way to handle ignoring some parts of the code - by doing so directly in the source code with, for example, [disable/enable rubocop comments](https://docs.rubocop.org/rubocop/configuration.html#disabling-cops-within-source-code) disabling cops, departments, or the entire rubocop, but since I've looked at rubocop in the wrong way it seemed superfluous back then. Now I know better.

It is now all fixed.

Thank you for the feedback!

katemildner committed:
add rubocop configuration file

katemildner committed:
resolve remaining style offenses
